jetty-core/jetty-http2/jetty-http2-client-transport/src/main/java/org/eclipse/jetty/http2/client/transport/internal/ClientHTTP2StreamEndPoint.java

```diff
@@ -63,7 +72,7 @@ public Runnable onTimeout(TimeoutException timeout, Promise<Boolean> promise)
             promise.succeeded(true);
             return null;
         }
-        return new Invocable.ReadyTask(Invocable.InvocationType.NON_BLOCKING, () ->
+        return () ->
```

Comment:
You changed the invocation type here because it's now executed immediately, right?

This behavior definitely should be documented somewhere.

Reply:
Yes, it's executed immediately.